src/es/manhwaweb/src/eu/kanade/tachiyomi/extension/es/manhwaweb/ManhwaWeb.kt

```diff
@@ -0,0 +1,191 @@
+
+    override val lang = "es"
+
+    override val supportsLatest = true
+
+    private val json: Json by injectLazy()
+
+    override val client: OkHttpClient = network.client.newBuilder()
+        .rateLimitHost(baseUrl.toHttpUrl(), 2)
+        .build()
+
+    override fun headersBuilder(): Headers.Builder = Headers.Builder()
+        .add("Referer", "$baseUrl/")
+
+    override fun popularMangaRequest(page: Int): Request = GET("$apiUrl/manhwa/nuevos", headers)
+
+    override fun popularMangaParse(response: Response): MangasPage {
+        val result = json.decodeFromString<PayloadPopularDto>(response.body.string())
+        val mangas = (result.data.weekly + result.data.total)
+            .distinctBy { it.slug }
+            .sortedByDescending { it.views }
+            .map { it.toSManga() }
+
+        return MangasPage(mangas, false)
+    }
+
+    override fun latestUpdatesRequest(page: Int): Request = GET("$apiUrl/latest/new-manhwa", headers)
+
+    override fun latestUpdatesParse(response: Response): MangasPage {
+        val result = json.decodeFromString<PayloadLatestDto>(response.body.string())
+        val mangas = (result.data.esp + result.data.raw18 + result.data.esp18)
+            .distinctBy { it.type + it.slug }
+            .sortedByDescending { it.latestChapterDate }
+            .map { it.toSManga() }
+
+        return MangasPage(mangas, false)
+    }
+
+    override fun searchMangaRequest(page: Int, query: String, filters: FilterList): Request {
+        val url = "$apiUrl/manhwa/library".toHttpUrl().newBuilder()
+            .addQueryParameter("buscar", query)
+
+        val typeFilter = filters.firstInstanceOrNull<TypeFilter>()
+        val demographyFilter = filters.firstInstanceOrNull<DemographyFilter>()
+        val statusFilter = filters.firstInstanceOrNull<StatusFilter>()
+        val eroticFilter = filters.firstInstanceOrNull<EroticFilter>()
+        val genreFilter = filters.firstInstanceOrNull<GenreFilter>()
+        val sortByFilter = filters.firstInstanceOrNull<SortByFilter>()
+
+        if (typeFilter != null) url.addQueryParameter("tipo", typeFilter.toUriPart())
+        if (demographyFilter != null) url.addQueryParameter("demografia", demographyFilter.toUriPart())
+        if (statusFilter != null) url.addQueryParameter("estado", statusFilter.toUriPart())
+        if (eroticFilter != null) url.addQueryParameter("erotico", eroticFilter.toUriPart())
+
+        if (genreFilter != null) {
+            val genres = genreFilter.state
+                .filter { it.state }
+                .joinToString("a") { it.id.toString() }
+            url.addQueryParameter("generes", genres)
+        }
+
+        url.addQueryParameter("order_dir", if (sortByFilter?.state?.ascending == true) "asc" else "desc")
+        url.addQueryParameter("order_item", sortByFilter?.selected ?: "alfabetico")
+
+        url.addQueryParameter("page", (page - 1).toString())
```

Comment:
just iterate on the filters and use `is TypeFilter` etc to build the url, this way no null checks.